### PR TITLE
[Editor] Disallow to have multiple pointers while dragging an editor

### PR DESCRIPTION
**Pull request by calixteman**
It'll let the user dragging with two fingers.

### CODE REVIEW

**Comment by calixteman** [comment]
Unfortunately the test isn't failing without the patch and the current version of Puppeteer we use, but with the last one (at least) it's fine.
/botio integrationtest

File: src/display/editor/editor.js [hunk 1]
```diff
@@ -1140,40 +1142,61 @@ class AnnotationEditor {
     const signal = this._uiManager.combinedSignal(ac);
```
Comment (Snuffleupagus) [inline src/display/editor/editor.js:1142]: How about reducing a bit of object creation below?
```suggestion
    const signal = this._uiManager.combinedSignal(ac);
    const opts = { capture: true, passive: false, signal };
```

File: src/display/editor/editor.js [hunk 2]
```diff
@@ -1140,40 +1142,61 @@ class AnnotationEditor {
...
+          );
+          this.#prevDragX = x;
+          this.#prevDragY = y;
+          this._uiManager.dragSelectedEditors(tx, ty);
+        },
+        {
+          passive: false,
+          capture: true,
+          signal,
+        }
```
Comment (Snuffleupagus) [inline src/display/editor/editor.js:1169]:
```suggestion
       opts
```

File: src/display/editor/editor.js [hunk 3]
```diff
@@ -1140,40 +1142,61 @@ class AnnotationEditor {
...
+      );
       window.addEventListener(
         "touchmove",
         e => {
           // Prevent the page from scrolling.
           e.preventDefault();
+          e.stopPropagation();
+        },
+        { capture: true, passive: false, signal }
```
Comment (Snuffleupagus) [inline src/display/editor/editor.js:1178]:
```suggestion
        opts
```

File: src/display/editor/editor.js [hunk 4]
```diff
@@ -1140,40 +1142,61 @@ class AnnotationEditor {
...
+      );
+      window.addEventListener(
+        "pointerdown",
+        e => {
+          // If the user drags with one finger and then clicks with another.
+          e.preventDefault();
+          e.stopPropagation();
         },
-        { passive: false, signal }
+        { capture: true, passive: false, signal }
```
Comment (Snuffleupagus) [inline src/display/editor/editor.js:1187]:
```suggestion
        opts
```

File: src/display/editor/editor.js [hunk 5]
```diff
@@ -1140,40 +1142,61 @@ class AnnotationEditor {
...
-        signal,
-      });
+      this.#dragPointerId = event.pointerId;
+      window.addEventListener(
+        "pointermove",
+        e => {
+          const { clientX: x, clientY: y, pointerId } = e;
+          if (pointerId !== this.#dragPointerId) {
+            e.preventDefault();
+            e.stopPropagation();
```
Comment (Snuffleupagus) [inline src/display/editor/editor.js:1154]: This pattern seems to be repeated a bunch of times below, how about introducing a small helper function and use that everywhere instead?

E.g.
```javascript
const stopEvent = e => {
  e.preventDefault();
  e.stopPropagation();
};
```
and we can change this code to
```suggestion
            stopEvent(e);
```

**Review by Snuffleupagus — APPROVED** [review]
r=me, with a few suggestions to reduce a little bit of duplication; thank you.

**Pull request merged**